### DISCUSSION TITLE
About HRRPUA and reference area

### DISCUSSION THREAD

**Discussion by abc-k** [firemodels/fds#12418] [Q&A]
I am trying to found out how the area referenced changes when the HRR is output.

In the user guide,

_"The OBST dimensions specified by XB in the input file may not (usually do not) perfectly align with the underlying Cartesian mesh. FDS then “snaps” the geometry to fill (or void) the volume of the nearest cell."_
_"If a specified mass flux boundary (including HRRPUA) is given, then FDS will automatically adjust this mass flux so that the total of the specified area times the specified mass flux is maintained._ (_user guide 6.8.0_ ,7.2.3  Specified Versus Actual Areas)"

I found the following statement.

Now I thought I want to specify the area by XB in the OBST line and set the HRRPUA in the SURF line.

Does this mean that the FDS changes **specified HRRPUA** to **adjusted HRRPUA** so that **specified area x specified HRRPUA = Area-delimited-by-mesh x adjusted HRRPUA**?

**Comment by drjfloyd** [comment]
Yes. If your input had an area of 1 m2 but because of grid snapping in the FDS domain it was 2 m2, then FDS would reduce the HRRPUA by a factor of 2 so you would get the desired total heat release.

**Reply by abc-k** [reply]
I appreciate your helping.
Is there a way to output HRRPUA and AREA adjusted by FDS?

**Reply by rmcdermo** [reply]
Yes, see the table at the end of the user guide for Solid Phase Output quantities.  There you will see that HRRPUA is an output.  Next, under the section on Spatially Integrated Outputs you will see SPATIAL_STATISTIC='AREA', which can be used to give the area of a surface.

**Reply by abc-k** [reply]
I appreciate your helping.
I used the Output quantities you taught me and created an input file as posted in the reply.
However, I could not output adjusted area for any of the XB, XYZ or SURF_ID inputs.

Is there anything else needed to configure the HRRPUA device?

**Reply by abc-k** [reply]
&HEAD CHID='HRRPUA', TITLE=''  /
&MESH IJK=24,6,18, XB=8.0,14.0,13.5,15,0.0,4.5/

&TIME T_END=500 /

&DUMP MASS_FILE=T /

&REAC ID = 'PROPANE'
	  FUEL='PROPANE'/      

!SURF 
&SURF ID='PROPANE',
      COLOR='RED',
      SPEC_ID='PROPANE'
      HRRPUA=290/

&VENT MB='XMIN', SURF_ID='OPEN' /
&VENT MB='XMAX', SURF_ID='OPEN' /
&VENT MB='YMIN', SURF_ID='OPEN' /
&VENT MB='YMAX', SURF_ID='OPEN' /
&VENT MB='ZMIN', SURF_ID='OPEN' /
&VENT MB='ZMAX', SURF_ID='OPEN' /


!BURN OBST
&OBST ID='EleCab',XB= 9.25, 11.65, 13.5,14.1, 0.3, 2.84, SURF_ID6='PROPANE','PROPANE',,'PROPANE',,'PROPANE'/

!DEVC
&DEVC ID='HRRPUA_TOP'    ,XB=  9.25, 11.65, 13.5,14.1, 2.84, 2.84,QUANTITY='HRRPUA',SPATIAL_STATISTIC='AREA'/
&DEVC ID='HRRPUA_L'      ,XB=  9.25,  9.25, 13.5,14.1, 0.30, 2.84,QUANTITY='HRRPUA',SPATIAL_STATISTIC='AREA'/
&DEVC ID='HRRPUA_R'      ,XB= 11.65, 11.65, 13.5,14.1, 0.30, 2.84,QUANTITY='HRRPUA',SPATIAL_STATISTIC='AREA'/
&DEVC ID='HRRPUA_F'      ,XB=  9.25, 11.65, 14.1,14.1, 0.30, 2.84,QUANTITY='HRRPUA',SPATIAL_STATISTIC='AREA'/
&DEVC ID='HRRPUA_TOTAL'  ,XB=  9.25, 11.65, 13.5,14.1, 0.30, 2.84,QUANTITY='HRRPUA',SPATIAL_STATISTIC='AREA'/

&DEVC ID='HRRPUA_POINT_TOP'    ,XYZ=10.00,14.0,2.84,QUANTITY='HRRPUA',SPATIAL_STATISTIC='AREA'/
&DEVC ID='HRRPUA_POINT_L'      ,XYZ= 9.25,14.0,2.00,QUANTITY='HRRPUA',SPATIAL_STATISTIC='AREA'/
&DEVC ID='HRRPUA_POINT_R'      ,XYZ=11.65,14.0,2.00,QUANTITY='HRRPUA',SPATIAL_STATISTIC='AREA'/
&DEVC ID='HRRPUA_POINT_F'      ,XYZ=10.00,14.1,2.00,QUANTITY='HRRPUA',SPATIAL_STATISTIC='AREA'/

&DEVC ID='S_HRRPUA_TOP'    ,XB=  9.25, 11.65, 13.5,14.1, 2.84, 2.84,QUANTITY='HRRPUA',SPATIAL_STATISTIC='AREA',SURF_ID='PROPANE'/
&DEVC ID='S_HRRPUA_L'      ,XB=  9.25,  9.25, 13.5,14.1, 0.30, 2.84,QUANTITY='HRRPUA',SPATIAL_STATISTIC='AREA',SURF_ID='PROPANE'/
&DEVC ID='S_HRRPUA_R'      ,XB= 11.65, 11.65, 13.5,14.1, 0.30, 2.84,QUANTITY='HRRPUA',SPATIAL_STATISTIC='AREA',SURF_ID='PROPANE'/
&DEVC ID='S_HRRPUA_F'      ,XB=  9.25, 11.65, 14.1,14.1, 0.30, 2.84,QUANTITY='HRRPUA',SPATIAL_STATISTIC='AREA',SURF_ID='PROPANE'/
&DEVC ID='S_HRRPUA_TOTAL'  ,XB=  9.25, 11.65, 13.5,14.1, 0.30, 2.84,QUANTITY='HRRPUA',SPATIAL_STATISTIC='AREA',SURF_ID='PROPANE'/

&DEVC ID='S_HRRPUA_POINT_TOP'    ,XYZ=10.00,14.0,2.84,QUANTITY='HRRPUA',SPATIAL_STATISTIC='AREA',SURF_ID='PROPANE'/
&DEVC ID='S_HRRPUA_POINT_L'      ,XYZ= 9.25,14.0,2.00,QUANTITY='HRRPUA',SPATIAL_STATISTIC='AREA',SURF_ID='PROPANE'/
&DEVC ID='S_HRRPUA_POINT_R'      ,XYZ=11.65,14.0,2.00,QUANTITY='HRRPUA',SPATIAL_STATISTIC='AREA',SURF_ID='PROPANE'/
&DEVC ID='S_HRRPUA_POINT_F'      ,XYZ=10.00,14.1,2.00,QUANTITY='HRRPUA',SPATIAL_STATISTIC='AREA',SURF_ID='PROPANE'/


&TALE
